### PR TITLE
Add Java and Python code for cascade classifier and HDR tutorials

### PR DESCRIPTION
**Pull request by catree**

### CODE REVIEW

**Comment by catree** [comment]
Fix `Debevec` typos in Python tutorials. Add link to download the data. Add additional resource links.

File: doc/py_tutorials/py_photo/py_hdr/py_hdr.markdown [hunk 1]
```diff
@@ -166,12 +166,12 @@ For this sequence we got the following estimation:
...
+1. Paul E Debevec and Jitendra Malik. Recovering high dynamic range radiance maps from photographs. In ACM SIGGRAPH 2008 classes, page 31. ACM, 2008. @cite DM97
+2. Mark A Robertson, Sean Borman, and Robert L Stevenson. Dynamic range improvement through multiple exposures. In Image Processing, 1999. ICIP 99. Proceedings. 1999 International Conference on, volume 3, pages 159–163. IEEE, 1999. @cite RB99
+3. Tom Mertens, Jan Kautz, and Frank Van Reeth. Exposure fusion. In Computer Graphics and Applications, 2007. PG'07. 15th Pacific Conference on, pages 382–390. IEEE, 2007. @cite MK07
 4. Images from [Wikipedia-HDR](https://en.wikipedia.org/wiki/High-dynamic-range_imaging)
 
 Exercises
 ---------
 1. Try all tonemap algorithms: [Drago](http://docs.opencv.org/master/da/d53/classcv_1_1TonemapDrago.html), [Durand](http://docs.opencv.org/master/da/d3d/classcv_1_1TonemapDurand.html), [Mantiuk](http://docs.opencv.org/master/de/d76/classcv_1_1TonemapMantiuk.html) and [Reinhard](http://docs.opencv.org/master/d0/dec/classcv_1_1TonemapReinhard.html).
```
Comment (alalek) [inline doc/py_tutorials/py_photo/py_hdr/py_hdr.markdown]: Could you please eliminate direct links to docs.opencv.org/master? (here and below)

This works pretty well:
```
Try all tonemap algorithms: cv::TonemapDrago, cv::TonemapDurand, cv::TonemapMantiuk and cv::TonemapReinhard
```

File: doc/py_tutorials/py_photo/py_hdr/py_hdr.markdown [hunk 2]
```diff
@@ -127,9 +127,9 @@ You can see the different results but consider that each algorithm have addition
 extra parameters that you should fit to get your desired outcome. Best practice is
 to try the different methods and see which one performs best for your scene.
 
-### Debvec:
+### Debevec:
 
-![image](images/ldr_debvec.jpg)
+![image](images/ldr_debevec.jpg)
```
Comment (alalek) [inline doc/py_tutorials/py_photo/py_hdr/py_hdr.markdown]: Is image file renamed?

Comment (catree) [reply doc/py_tutorials/py_photo/py_hdr/py_hdr.markdown]: Indeed, I have renamed the image.

**Pull request merged**